js/packages/cli/src/commands/set-collection.ts

```diff
@@ -157,7 +157,7 @@ export async function setCollection(
   }
 
   instructions.push(
-    await anchorProgram.instruction.setCollection({
+    anchorProgram.instruction.setCollection({
```

Comment:
The documentation shows this as an async function. Are they incorrect? 
<img width="678" alt="image" src="https://user-images.githubusercontent.com/41593264/164547520-7e4b856b-49ce-4b3b-a558-2d91868217de.png">

Reply:
`metaplex-program-library\token-metadata\js\src\generated\instructions`
`import {  CreateMasterEditionV3,} from '@metaplex-foundation/mpl-token-metadata';`

Compared to:

`metaplex-program-library\candy-machine\js\src\generated\instructions

import {   setCollection, } from '@metaplex-foundation/mpl-candy-machine;`
both should be functions based on the definition, but the return is 

`anchorProgram.instruction.setCollection`
`  const ix = new web3.TransactionInstruction({`
`    programId: new web3.PublicKey('cndy3Z4yapfJBmL3ShUp5exZKqR3z33thTzeNMm2gRZ'),`
`    keys,`
`    data,`
`  });`
  return ix;`
  
`  CreateMasterEditionV3`
`    const ix = new web3.TransactionInstruction({`
`    programId: new web3.PublicKey('metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s'),`
`    keys,`
`    data,`
`  });`
`  return ix;`

So with that comparison, I would have to answer that it is incorrect, because although it is defined as a function, it does not execute code, but rather returns an instruction to be executed.